### PR TITLE
NPE while processing PRAGMA compiler directive

### PR DESCRIPTION
**Pull request by lananda**
Signed-off-by: Lalitha Ananda <lalitha.ananda@oracle.com>

### CODE REVIEW

**Comment by lananda** [comment]
If a PL/SQL procedure declaration has PRAGMA compiler tag, the DDL parser is failing to parser this & throws NPE

e.g.
create or replace package azlwbs.azlwbs_jos
is

PRAGMA SERIALLY_REUSABLE; --> NOT GETTING PARSED
                                 
  procedure wsazl0(  azl_request varchar2, response out varchar2 );

end azlwbs_jos;
/

DDLParser.jjt has the below logic: for packageDeclaration processing it's having switch case for cursorDeclaration,functionSpec,procedureSpec but the pragmaDeclaration case is inside the default case & it never reaches the else part of the if else block in default case. I've moved it above default case.

final public void packageDeclaration(PLSQLPackageType packageType) throws ParseException {
    switch ((jj_ntk==-1)?jj_ntk():jj_ntk) {
    case R_SUBTYPE:
    case R_TYPE: <snip>
    case R_CURSOR:
    case R_PROCEDURE: <snip>
    case R_FUNCTION: <snip>
    default:
      if (jj_2_3(2)) {
        variableDeclaration(packageType);
      } else {
        switch ((jj_ntk==-1)?jj_ntk():jj_ntk) {
        case K_PRAGMA:
          pragmaDeclaration();
          break;
        default:
          jj_consume_token(-1);
          throw new ParseException();
        }
      }
    }
}


Original:
void packageDeclaration(PLSQLPackageType packageType) :
{}
{
      typeOrSubTypeDeclaration(packageType)
    | cursorDeclaration(packageType)
    | procedureSpec(packageType)
    | functionSpec(packageType)
    | LOOKAHEAD(2) variableDeclaration(packageType) -- FAILS HERE
    | pragmaDeclaration()
}

Proposed FIX:
void packageDeclaration(PLSQLPackageType packageType) :
{}
{
      typeOrSubTypeDeclaration(packageType)
    | cursorDeclaration(packageType)
    | procedureSpec(packageType)
    | functionSpec(packageType)
    | pragmaDeclaration()
    <O_SEMICOLON>
    | LOOKAHEAD(2) variableDeclaration(packageType)
}

**Comment by lananda** [comment]
[ddlparser_bug30265782.zip](https://github.com/eclipse-ee4j/eclipselink-oracleddlparser/files/4374679/ddlparser_bug30265782.zip)

Reproducer

File: oracleddlparser/src/main/jjtree/org/eclipse/persistence/tools/oracleddl/parser/DDLParser.jjt [hunk 1]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2011, 2018 Oracle and/or its affiliates. All rights reserved.
+ * Copyright (c) 2020 Oracle and/or its affiliates. All rights reserved.
```
Comment (lukasj) [inline oracleddlparser/src/main/jjtree/org/eclipse/persistence/tools/oracleddl/parser/DDLParser.jjt]: should be "2011, 2020"

**Comment by lananda** [comment]
> LGTM

Thanks Lukas.

**Comment by lananda** [comment]
@lukasj Lukas, would you please let me know how to integrate the latest oracleddlparser.jar with eclipselink project?

**Pull request merged**